pkg/java/pom/parse.go

```diff
@@ -86,6 +88,24 @@ func (p *parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 		return nil, nil, xerrors.Errorf("failed to parse POM: %w", err)
 	}
 
+	for _, rep := range content.Repositories.Repository {
+		repoURL, err := url.Parse(rep.URL)
+		if err != nil {
+			continue
+		}
+
+		for _, server := range p.settings.Servers {
+			if rep.ID == server.ID && server.Username != "" && server.Password != "" {
+				repoURL.User = url.UserPassword(server.Username, server.Password)
+				break
+			}
+		}
+
+		if rep.Releases.Enabled != "false" {
+			p.remoteRepositories = append(p.remoteRepositories, repoURL.String())
+		}
```

Comment:
It looks like we can't use repositories only for enabled `releases` - https://maven.apache.org/settings.html#repositories.
IIUC we need to check `pluginRepositories` along with `releases`.

But this looks like rare case. Perhaps we can start by adding all repositories.

Reply:
Added pluginRepositories also

Reply:
hm... i re-read maven docs and it seems I made mistake and confused you.
`pluginRepositories` is used only for set up repositories to downloading plugins.
 But we don't check plugins, so we don't need this.

But I thought about “snapshots.” What if we add a repository if `release` or `snapshot` is enabled:
```go
for _, rep := range content.Repositories.Repository {
		if rep.Releases.Enabled == "false" || rep.Snapshots.Enabled == "false" {
			continue
		}

		repoURL, err := url.Parse(rep.URL)
		if err != nil {
			continue
		}

		for _, server := range p.servers {
			if rep.ID == server.ID && server.Username != "" && server.Password != "" {
				repoURL.User = url.UserPassword(server.Username, server.Password)
				break
			}
		}

		log.Logger.Debugf("Adding repository %s: %s", rep.ID, rep.URL)
		p.remoteRepositories = append(p.remoteRepositories, repoURL.String())
	}
```